packages/editor-ui/src/components/executions/workflow/WorkflowExecutionAnnotationPanel.vue

```diff
@@ -19,125 +35,75 @@ const hasChanged = (prev: string[], curr: string[]) => {
-		},
-		loading: {
-			type: Boolean,
-			default: true,
-		},
-	},
-
-	computed: {
-		...mapStores(useExecutionsStore, useWorkflowsStore),
-		vote() {
-			return this.activeExecution?.annotation?.vote || null;
-		},
-		activeExecution() {
-			// FIXME: this is a temporary workaround to make TS happy. activeExecution may contain customData, but it is type-casted to ExecutionSummary after fetching from the backend
-			return this.executionsStore.activeExecution as ExecutionSummary & {
-				customData?: Record<string, string>;
-			};
-		},
-		tagIds() {
-			return this.activeExecution?.annotation?.tags.map((tag) => tag.id) ?? [];
-		},
-		tags() {
-			return this.activeExecution?.annotation?.tags;
-		},
-	},
-	setup() {
-		return {
-			...useToast(),
-		};
-	},
-	data() {
-		return {
-			tagsEventBus: createEventBus(),
-			isTagsEditEnabled: false,
-			appliedTagIds: [] as string[],
-			tagsSaving: false,
-		};
-	},
-	methods: {
-		async onVoteClick(vote: AnnotationVote) {
-			if (!this.activeExecution) {
-				return;
-			}
-
-			// If user clicked on the same vote, remove it
-			// so that vote buttons act as toggle buttons
-			const voteToSet = vote === this.vote ? null : vote;
-
-			try {
-				await this.executionsStore.annotateExecution(this.activeExecution.id, { vote: voteToSet });
-			} catch (e) {
-				this.showError(e, this.$locale.baseText('executionAnnotationView.vote.error'));
-			}
-		},
-		onTagsEditEnable() {
-			this.appliedTagIds = this.tagIds;
-			this.isTagsEditEnabled = true;
-
-			setTimeout(() => {
-				this.tagsEventBus.emit('focus');
-			}, 0);
-		},
-		async onTagsBlur() {
-			if (!this.activeExecution) {
-				return;
-			}
-
-			const current = (this.tagIds ?? []) as string[];
-			const tags = this.appliedTagIds;
-
-			if (!hasChanged(current, tags)) {
-				this.isTagsEditEnabled = false;
-				return;
-			}
-
-			if (this.tagsSaving) {
-				return;
-			}
-
-			this.tagsSaving = true;
-
-			try {
-				await this.executionsStore.annotateExecution(this.activeExecution.id, { tags });
-			} catch (e) {
-				this.showError(e, this.$locale.baseText('executionAnnotationView.tag.error'));
-			}
-
-			this.tagsSaving = false;
-			this.isTagsEditEnabled = false;
-		},
-		onTagsEditEsc() {
-			this.isTagsEditEnabled = false;
-		},
-	},
-});
+const onVoteClick = async (voteValue: AnnotationVote) => {
+	if (!activeExecution.value) {
+		return;
+	}
+
+	const voteToSet = voteValue === vote.value ? null : voteValue;
+
+	try {
+		await executionsStore.annotateExecution(activeExecution.value.id, { vote: voteToSet });
+	} catch (e) {
+		showError(e, 'executionAnnotationView.vote.error');
+	}
+};
+
+const onTagsEditEnable = () => {
+	appliedTagIds.value = tagIds.value;
+	isTagsEditEnabled.value = true;
+
+	setTimeout(() => {
+		tagsEventBus.emit('focus');
+	}, 0);
+};
+
+const onTagsBlur = async () => {
+	if (!activeExecution.value) {
+		return;
+	}
+
+	const current = (tagIds.value ?? []) as string[];
+	const tags = appliedTagIds.value;
```

Comment:
Linter warning here: `'tags' is already declared in the upper scope on line 27 column 7.eslint[@typescript-eslint/no-shadow](https://typescript-eslint.io/rules/no-shadow)`